server/proxy/proxy.go

```diff
@@ -178,3 +187,67 @@ func (p *Proxy) upstreamForApp(app *apps.App) (upstream.Upstream, error) {
 		return nil, utils.NewInvalidError("not a valid app type: %s", app.AppType)
 	}
 }
+
+func (p *Proxy) CleanUserCallContext(userID string, cc *apps.Context) (*apps.Context, error) {
```

Comment:
As pointed out below I think it should be a simple `CheckCallContext(userID string, cc *apps.Context) error` function.

Reply:
@hanzei How can we discard unwanted fields if we only return an error?

Reply:
Just overwrite the struct behind the pointer.